gobblin-api/src/main/java/org/apache/gobblin/configuration/State.java

```diff
@@ -500,6 +501,30 @@ public void removeProp(String key) {
   }
 
   /**
+   * Remove all properties with a certain keyPrefix
+   *
+   * @param keyPrefix key prefix
+   */
+  public void removeProps(String keyPrefix) {
```

Comment:
Maybe rename this to `removePropsWithPrefix`? The current name is too easy to mess up as a typo.

Reply:
Good point.